src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt

```diff
@@ -194,6 +193,22 @@ fun Ty.isMovesByDefault(lookup: ImplLookup): Boolean =
         else -> !lookup.isCopy(this)
     }
 
+/**
+ * If `Ty` is a `Future` or convertible into a `Future` (i.e. `IntoFuture`), get the type of
+ * that future's output.
+ *
+ * If `Ty` is not convertible into a future, or if the `Future` and `IntoFuture` lang items are missing,
+ * returns `TyUnknown`.
+ *
+ * The `IntoFuture` trait is automatically used for Rust versions which support it.
+ */
+fun Ty.lookupFutureOutputTy(lookup: ImplLookup): TyWithObligations<Ty> {
+    val futureTrait = lookup.items.IntoFuture ?: lookup.items.Future ?: return TyWithObligations(TyUnknown)
+    val outputType = futureTrait.findAssociatedType("Output") ?: return TyWithObligations(TyUnknown)
+    val selection = lookup.selectProjection(TraitRef(this, futureTrait.withSubst()), outputType)
```

Comment:
@vlad20012 The version from `RsAwaitCompletionProvider` uses `selectProjectionStrict`. Can we use `selectProjection` here?